Backend/src/test/java/project/airbnb/reservation/ReservationControllerTest.java

```diff
@@ -17,68 +20,118 @@
+			50_000L, new BnbOption(1, 2, 1),
+			new CheckInOutTime(LocalTime.of(15, 0), LocalTime.of(11, 0)),
+			BnbType.HOTEL, "숙소 설명", "호스트명", 3,
+			4.5d, 200);
+
+		BnbImage bnbImage1 = new BnbImage(null, null, "http://www.naver111.com");
+		BnbImage bnbImage2 = new BnbImage(null, null, "http://www.naver222.com");
+		bnb.saveBnbImage(bnbImage1);
+		bnb.saveBnbImage(bnbImage2);
 		bnbRepository.save(bnb);
 
-		Member member = new Member(null, new ArrayList<>());
+		Member member = new Member(null, "nori");
+		Member foundMember = memberRepository.save(member);
+		memberId = foundMember.getId();
+
+		Reservation reservation1 = new Reservation(null, member, bnb,
+			new CheckInOutDate(LocalDate.of(2022, 1, 1),
+				LocalDate.of(2022, 1, 6)),
+			new Headcount(2, 2, 0), 800_000L);
+		Reservation save1 = reservationRepository.save(reservation1);
+		reservationId1 = save1.getId();
+
+		Reservation reservation2 = new Reservation(null, member, bnb,
+			new CheckInOutDate(LocalDate.of(2022, 3, 3),
+				LocalDate.of(2022, 3, 5)),
+			new Headcount(2, 0, 0), 200_000L);
+		Reservation save2 = reservationRepository.save(reservation2);
+		reservationId2 = save2.getId();
+	}
+
+	@Test
+	@DisplayName("showList 호출 시 지정된 멤버의 모든 예약 내역이 반환된다")
+	public void showReservationList() {
+		// given
+
+		// when
+		ApiResponse<List<ShortReservationDto>> response = reservationController.showList(
+			memberId);
+		List<ShortReservationDto> list = response.getData();
+		for (ShortReservationDto dto : list) {
+			log.debug("ShortReservationDto : {}", dto);
+		}
+		ShortReservationDto dto1 = list.get(0);
+		ShortReservationDto dto2 = list.get(1);
 
-		Reservation reservation = new Reservation(null, member, null,
-			LocalDate.of(2022, 1, 1),
-			LocalDate.of(2022, 1, 6),
-			2, 2, 0, 800_000L);
-		reservation.setBnb(bnb);
-		reservation.saveMember(member);
+		// then
+		assertThat(list.size()).isEqualTo(2);
+		assertThat(dto1.getReservationId()).isEqualTo(reservationId1);
+		assertThat(dto1.getBnbName()).isEqualTo("숙소 이름");
+		assertThat(dto1.getImageUrl()).isEqualTo("http://www.naver111.com");
 
-		reservationRepository.save(reservation);
-		memberRepository.save(member);
+		assertThat(dto2.getReservationId()).isEqualTo(reservationId2);
+		assertThat(dto2.getBnbName()).isEqualTo("숙소 이름");
+		assertThat(dto2.getImageUrl()).isEqualTo("http://www.naver111.com");
 
 	}
 
 	@Test
-	@DisplayName("회원이 저장되나")
-	public void saveMember() {
+	@DisplayName("showDetails 호출 시 지정된 id의 LongReservationDto가 반환된다")
+	public void showReservationDetails() {
+
 		// given
 
 		// when
-		List<SimpleReservationDto> list = reservationController.list();
-		System.out.println("=================================");
-		list.forEach(System.out::println);
-		System.out.println("=================================");
+		ApiResponse<LongReservationDto> response = reservationController.showDetails(
+			reservationId1);
+		LongReservationDto dto = response.getData();
+		log.debug("LongReservationDto : {}", dto);
```

Comment:
when, then을 좀 더 구분할 수 있을 것 같습니다.

Reply:
아래 방식으로 바꿔보았습니다. 이런식으로 구분하는걸 말씀하신걸까요? 👀

```suggestion
		// when
		ApiResponse<LongReservationDto> response = reservationController.showDetails(
			reservationId1);
		// then 
     assertThat(response.getData()).isExactlyInstanceOf(LongReservationDto.class);
		...
```

Reply:
네 맞습니다.

- response가 만들어지는 것 까지가 when이고
- 만들어진 reponse에서 변수를 뽑아내는 것은 then의 영역입니다.